src/term/terminal.cpp

```diff
@@ -0,0 +1,103 @@
+#include "term/terminal.h"
+
+#include <stdexcept>
+
+extern "C" {
+#include <stdio.h>
+#include <sys/ioctl.h>
+#include <termios.h>
+#include <unistd.h>
+}
+
+const char* TERMINAL = "/dev/tty";
+
+struct TerminalImpl {
+    FILE *tty;
+    int fd;
+    struct termios oldTermios, newTermios;
+};
```

Comment:
Used the pimpl (private implementation pattern) to hide the C headers and unix specific includes from all the users of the library.